src/Lecture2.hs

```diff
@@ -150,14 +170,219 @@ You're free to define any helper functions.
        treasure besides gold (if you already haven't done this).
 -}
 
--- some help in the beginning ;)
+newtype Gold = Gold Int
+newtype XP = XP Int
+newtype Health = Health Int deriving (Eq, Ord)
+newtype Attack = Attack Int deriving (Eq, Ord)
+newtype Endurance = Endurance Int deriving (Eq, Ord)
+
+instance Num Health where
+    (+) (Health a) (Health b) = Health (a + b)
+    (*) (Health a) (Health b) = Health (a * b)
+    abs (Health x) = Health (abs x)
+    signum (Health x) = Health (signum x)
+    (-) (Health a) (Health b) = Health (a - b)
+    fromInteger = Health . fromInteger
+
+instance Num Attack where
+    (+) (Attack a) (Attack b) = Attack (a + b)
+    (*) (Attack a) (Attack b) = Attack (a * b)
+    abs (Attack x) = Attack (abs x)
+    signum (Attack x) = Attack (signum x)
+    (-) (Attack a) (Attack b) = Attack (a - b)
+    fromInteger = Attack . fromInteger
+
+instance Num Endurance where
+    (+) (Endurance a) (Endurance b) = Endurance (a + b)
+    (*) (Endurance a) (Endurance b) = Endurance (a * b)
+    abs (Endurance x) = Endurance (abs x)
+    signum (Endurance x) = Endurance (signum x)
+    (-) (Endurance a) (Endurance b) = Endurance (a - b)
+    fromInteger = Endurance . fromInteger
```

Comment:
I haven't explained this yet. But all this boilerplate actually can be removed. If you add the following line in the beginning of the file:

```haskell
{-# LANGUAGE GeneralizedNewtypeDeriving #-}
```

You'll be able to derive `Num` for `Health`, `Attack` and `Endurance` as you already did with `Eq` and `Ord ` 👍🏻

Reply:
I was sure that there was a way to derive `Num` for `Num` newtypes, thanks